### PR TITLE
MAKE-1045: better test wire protocol with different wire things

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: wire/client_test.go [hunk 1]
```diff
@@ -290,22 +338,42 @@ func TestWireProcess(t *testing.T) {
...
+		Options testutil.OptsModify
+	}{
+		{
+			Name: "Blocking",
+			Options: func(opts *options.Create) {
+				opts.Implementation = options.ProcessImplementationBlocking
+			},
+		},
+		{
+			Name: "Blocking",
```
Comment (Kimchelly) [inline wire/client_test.go]: Duplicate blocking - there should be a basic case.

File: wire/client_test.go [hunk 2]
```diff
@@ -290,22 +338,42 @@ func TestWireProcess(t *testing.T) {
...
+			Options: func(opts *options.Create) {
+				opts.Implementation = options.ProcessImplementationBasic
+			},
+		},
+		{
+			Name:    "Default",
+			Options: func(opts *options.Create) {},
```
Comment (Kimchelly) [inline wire/client_test.go]: This is unnecessary.

File: wire/client_test.go [hunk 3]
```diff
@@ -24,17 +24,37 @@ func TestWireManager(t *testing.T) {
...
+		Options testutil.OptsModify
+	}{
+		{
+			Name: "Blocking",
+			Options: func(opts *options.Create) {
+				opts.Implementation = options.ProcessImplementationBlocking
+			},
+		},
+		{
+			Name: "Blocking",
```
Comment (Kimchelly) [inline wire/client_test.go]: Copy paste.

File: wire/client_test.go [hunk 4]
```diff
@@ -24,17 +24,37 @@ func TestWireManager(t *testing.T) {
...
+			Options: func(opts *options.Create) {
+				opts.Implementation = options.ProcessImplementationBasic
+			},
+		},
+		{
+			Name:    "Default",
```
Comment (Kimchelly) [inline wire/client_test.go]: Copy paste.

File: wire/client_test.go (hunk 2 above)
Comment (tychoish) [reply wire/client_test.go]: I know, but making sure that you don't need to set implementation is probably a good thing?

**Comment by tychoish** [comment]
it's my intention to merge rpc/rest/cli/wire into a single "remote" package (with jasper.Remote) and refactor the tests to use a single set of cases, and I'd like to do that as part of a seperate commit, as part of MAKE-1043, using this model.

File: wire/client_test.go (hunk 2 above)
Comment (Kimchelly) [reply wire/client_test.go]: This should be left to validation in create options.

Comment (tychoish) [reply wire/client_test.go]: right but if we're always setting it, then we don't know if the default operation works in this configuration.

Comment (Kimchelly) [reply wire/client_test.go]: I guess?

**Pull request merged**